powerdnsadmin/routes/index.py

```diff
@@ -684,6 +700,31 @@ def register():
         return render_template('errors/404.html'), 404
 
 
+# Show welcome page on first login if otp_force is enabled
+@index_bp.route('/welcome', methods=['GET', 'POST'])
+def welcome():
+    if 'welcome' not in session:
+        return redirect(url_for('index.index'))
+
+    user_id = session['user_id']
+    user = User(id = user_id)
+    login_user(user)
+    encoded_img_data = base64.b64encode(qrcode()[0])
```

Comment:
Could you take a look at `routes.user.qrcode()`? There are these three lines of code:
```python3
img = qrc.make(current_user.get_totp_uri(),
               image_factory=qrc_svg.SvgPathImage)
stream = BytesIO()
img.save(stream)
```
Can you move these into a separate function inside `models/user.py` (return `stream.getvalue()`), which you then call from within `routes.user.qrcode()` and in the line marked here? Then you don't have to call `login_user(user)` here just to be able to fetch the qrcode from the view function.

 Doing this would be fine because the user is already authenticated since `welcome_user_id` has been placed into the session after the user was either authenticated or registered.